### ISSUE TITLE
[Proposal] Хранение данных в mySQL

### ISSUE THREAD

**Issue by vasyaabr** [sportorg/pysport#325]
Привет!

А не было желания хранить базу данных соревнования в mySQL базе? Я думаю, многие связанные с IT организаторы и их помощники будут очень рады такой возможности. 

Преимущества следующие:

1. Простая организация многопользовательской работы. В mySQL давно всё решено с конкретным доступом,это вам не файл, в котором непонятно что произойдет при попытке конкурентной записи. Установил на любом компе сервер mysql и готово, все клиенты могут работать.

2. Очень удобно для всяческой автоматизации. Я сейчас провожу соревнования и тренировки используя MeOS, и мне очень удобно во-первых, выдергивать информацию в произвольном виде запросами из БД; во-вторых, прикручивать автоматизацию к уже лежащей в интернете БД (сплиты например делать: https://opskov.ru/cupcalc/observer.php?type=split&db=meos_20190105_210221_248&group=1, опубликовал базу из MeOS, хоп, сплиты уже есть).

**Comment by daxartio** [comment]
Привет. Конечный пользователь привык работатать с базой внутри файла (копирование, передача). SQL (mysql) база данных хороша для серверных программ. Для по SportOrg mysql избыточен.
1. Для многопользовательской работы у нас предполагается использовать "Командую работу". 

2 Для автоматизации у нас можно создать свои шаблоны на js, html, docx, или реализовать свои скрипты на питоне для передачи данных на сервер. Да и сам режим "Командая работа" это просто tcp соединение с json форматом.

Скорее всего мы можем дать возможность сохранять соревнования в mysql, но мапингом моделей никто не захочет заниматься.

**Comment by vasyaabr** [comment]
Если у меня найдется время покодить, вы не будете возражать против такой фичи (выбора способа хранения)? Я полистал ваш код, в принципе, объем разработки не слишком страшен.

**Comment by daxartio** [comment]
Да, конечно. Мы рады PR-ам. Главное, чтобы в дальнейшем это не требовало жесткой поддержки. Я предлагаю реализовать механизм записи/чтения в бд вместо файла. В дальнейшем, чтобы от нас требовалось только исправлять или дополнять модель БД, а не сам механизм.

**Comment by livello** [comment]
@vasyaabr @daxartio 
База данных - это только способ внутреннего хранения и обработки данных для программы. Использовать БД для взаимодействия нескольких программ - попахивает множеством костылей.
Только если по-другому ну никак или взаимодействие сильно ограниченное - как у вам, вытаскивание сплитов. 
В целом мотивация понятна. тот же мгновенный протокол. 
Но, ИМХО, правильнее было бы сделать ядро программы с интерфейсом взаимодействия, к которому могли бы подключаться другие клиенты.
Например, для приема оплаты и выдачи номеров и чипов уже зарегистрированным участникам...

**Comment by unaimillan** [comment]
Возможно, можно также использовать PostgreSQL

**Comment by vasyaabr** [comment]
SQL-драйвер, строго говоря, без разницы. Данных немного, оптимизации с учетом особенностей СУБД не требуется, запросы будут почти универсальными (или совсем, если использовать подходящую либу).
